assets/js/modules/adsense/components/setup/v2/SetupAccount.js

```diff
@@ -21,14 +21,89 @@
  */
 import PropTypes from 'prop-types';
 
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_ADSENSE } from '../../../datastore/constants';
+import {
+	ACCOUNT_STATUS_NO_CLIENT,
+	ACCOUNT_STATUS_PENDING_TASKS,
+	ACCOUNT_STATUS_APPROVED,
+	SITE_STATUS_NONE,
+	SITE_STATUS_NEEDS_ATTENTION,
+	determineClientID,
+} from '../../../util/status';
+import SetupAccountSite from './SetupAccountSite';
+import SetupAccountNoClient from './SetupAccountNoClient';
+import SetupAccountCreateSite from './SetupAccountCreateSite';
+import SetupAccountPendingTasks from './SetupAccountPendingTasks';
+const { useSelect, useDispatch } = Data;
+
 export default function SetupAccount( { account } ) {
-	return (
-		<div>TODO: UI to proceed with steps for account { account._id }</div>
+	const { _id: accountID, pendingTasks } = account;
+
+	const clientID = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getClientID()
+	);
+
+	const clients = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getClients( accountID )
+	);
+
+	const site = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getCurrentSite( accountID )
 	);
+
+	const acfClientID = determineClientID( { clients: clients || [] } );
```

Comment:
I noticed that this function returns `undefined` if there are no clients as well as if `clients` are `undefined`/loading. It might make more sense to change this to return `null` if there are none to differentiate below so that we don't show `SetupAccountNoClient` just because clients are loading. Unless we can assume that clients will always be loaded at this point, then I suppose it's fine; it's just a little inconsistent I feel with how `undefined` is used in most other places.

Reply:
@aaemnnosttv I agree with you but I don't want to change that function because it is already used in a couple of other places. I have left it as is and updated the SetupAccount component to display a progress bar until clients are loaded.